contrib/pg_tde/src/pg_tde_event_capture.c

```diff
@@ -58,7 +59,11 @@ checkEncryptionClause(const char *accessMethod)
 	if (tdeCurrentCreateEvent.encryptMode)
 	{
 		LWLockAcquire(tde_lwlock_enc_keys(), LW_SHARED);
-		principal_key = GetPrincipalKey(MyDatabaseId, LW_SHARED);
+		principal_key = GetPrincipalKeyNoDefault(MyDatabaseId, LW_SHARED);
+		if (principal_key == NULL)
+		{
+			principal_key = GetPrincipalKeyNoDefault(DEFAULT_DATA_TDE_OID, LW_EXCLUSIVE);
+		}
```

Comment:
Hmm, this seems a bit flakey. Shouldn't we instead make sure to always copy the default with an event trigger on `CREATE DATABASE`? Especially when think in the long term for when we add catalog encryption. Or would that require copying it even earlier in the process and therefore something we should look into later?

Reply:
And that wouldn't help with existing databases.